src/gui/sidebar/previews/layer/SidebarLayersContextMenu.cpp

```diff
@@ -0,0 +1,53 @@
+#include "gui/sidebar/previews/layer/SidebarLayersContextMenu.h"
+
+SidebarLayersContextMenu::SidebarLayersContextMenu(GladeGui* gui, SidebarToolbar* toolbar):
+        SidebarBaseContextMenu(gui->get("sidebarPreviewLayersContextMenu")) {
+
+    // Connect the context menu actions
+    const std::map<std::string, SidebarActions> ctxMenuActions = {
+            {"sidebarPreviewLayerMoveUp", SIDEBAR_ACTION_MOVE_UP},
+            {"sidebarPreviewLayerMoveDown", SIDEBAR_ACTION_MOVE_DOWN},
+            {"sidebarPreviewMergeDown", SIDEBAR_ACTION_MERGE_DOWN},
+            {"sidebarPreviewLayerDuplicate", SIDEBAR_ACTION_COPY},
+            {"sidebarPreviewLayerDelete", SIDEBAR_ACTION_DELETE},
+            // TODO(ja-he): add SIDEBAR_ACTION_NEW_BEFORE/AFTER if/when these are defined for layers
+    };
+
+
+    for (const auto& [name, action]: ctxMenuActions) {
+
+        GtkWidget* const entry = gui->get(name);
+        g_assert(entry != nullptr);
+
+        // Unfortunately, we need a fairly complicated mechanism to keep track
+        // of which action we want to execute.
```

Comment:
It might make more sense here to describe the code below. For example,
```suggestion
        // Create callbacks, store the entry
```